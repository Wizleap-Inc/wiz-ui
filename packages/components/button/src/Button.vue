<script setup lang="ts">
import { defineProps, onMounted, toRefs } from "vue";

interface Props {
  logText: string;
}

const props = defineProps<Props>();

const { logText } = toRefs(props);

const handleClick = (e: MouseEvent) => {
  e.preventDefault();
  console.log("clicked");
  console.log(logText);
};

onMounted(() => {
  console.log("mounted");
});
</script>

<template>
  <button class="button" @click="handleClick">button</button>
</template>

<style lang="scss" scoped>
.button {
  background-color: #111111;
  border-radius: 0.25em;
  border: none;
  color: #ffffff;
  cursor: pointer;
  font-size: 1.5em;
}
</style>
