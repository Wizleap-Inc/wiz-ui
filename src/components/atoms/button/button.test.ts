import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Button from "./button.vue";

describe("Button", () => {
  describe("Mount", () => {
    it("正常にレンダリングされるか", () => {
      const wrapper = mount(Button);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe("Slot", () => {
    it("テキストが正しく表示されるか", () => {
      const slotText = "Hello World!";
      const wrapper = mount(Button, {
        slots: {
          default: slotText,
        },
      });
      expect(wrapper.html()).toContain(slotText);
    });
  });

  describe("Props", () => {
    it("disabledが正しく設定されるか", () => {
      const wrapper = mount(Button, {
        propsData: {
          disabled: true,
        },
      });
      expect(wrapper.classes()).toContain("wiz-button--disabled");
    });
  });

  describe("Emits", () => {
    it("clickイベントが発火されるか", async () => {
      const wrapper = mount(Button);
      const button = wrapper.find("button");
      await button.trigger("click");
      expect(button.emitted("click")).toBeTruthy();
    });

    it("disabled時clickイベントが発火されないか", async () => {
      const wrapper = mount(Button, {
        propsData: {
          disabled: true,
        },
      });
      const button = wrapper.find("button");
      await button.trigger("click");
      expect(button.emitted("click")).toBeFalsy();
    });
  });
});
