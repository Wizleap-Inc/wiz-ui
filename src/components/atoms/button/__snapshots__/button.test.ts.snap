// Vitest Snapshot v1

exports[`Button > Mount > 正常にレンダリングされるか 1`] = `"<button data-v-c2b9c338=\\"\\" class=\\"wiz-button wiz-button--variant-primary wiz-button--size-md\\" style=\\"--c2b9c338-padding: 0.75rem 1.5rem; --c2b9c338-borderRadiusXxs: 0.25rem; --c2b9c338-colorGradient: linear-gradient(90deg, #3db783 0%, #099ec3 100%); --c2b9c338-colorWhite: #FFFFFF; --c2b9c338-shadowMd: 0px 0px 4px #00000040; --c2b9c338-colorGreen: #3DB783; --c2b9c338-gray400: #D3D8D7; --c2b9c338-borderRadiusMax: 9999px; --c2b9c338-fontSizeXs: 0.75rem; --c2b9c338-fontSizeSm: 0.875rem; --c2b9c338-fontSizeMd: 1rem;\\"></button>"`;

exports[`GradientButton > Mount > 正常にレンダリングされるか 1`] = `"<button data-v-c2b9c338=\\"\\" class=\\"wiz-button wiz-button--variant-primary wiz-button--size-md\\" style=\\"--c2b9c338-padding: 0.75rem 1.5rem; --c2b9c338-borderRadiusXxs: 0.25rem; --c2b9c338-colorGradient: linear-gradient(90deg, #3db783 0%, #099ec3 100%); --c2b9c338-colorWhite: #FFFFFF; --c2b9c338-shadowMd: 0px 0px 4px #00000040; --c2b9c338-colorGreen: #3DB783; --c2b9c338-gray400: #D3D8D7; --c2b9c338-borderRadiusMax: 9999px; --c2b9c338-fontSizeXs: 0.75rem; --c2b9c338-fontSizeSm: 0.875rem; --c2b9c338-fontSizeMd: 1rem;\\"></button>"`;
